File: Clases/geometria/circulo.py
from punto import Punto
import math

class Circulo(Punto):
    def __init__(self, x, y, radio):
        super().__init__(x, y)
        self.radio = abs(radio)

    def obterDiametro(self):
        return self.radio * 2

    def calcularPerimetro(self):
        return 2 * math.pi * self.radio

    def calcularArea(self):
        return math.pi * pow(self.radio, 2)
       #return math.pi * self.radio * self.radio
       #return math.pi * (self.radio ** 2)

    def aCadea(self):
        cadea = super().aCadea() + "\n\t Radio: " + str(self.radio) + "\n\t Diametro:"+ str(self.obterDiametro()) +"\n\t Area: " + str(self.calcularArea())
        return cadea

File: Java/proyectosJava/src/Circulo.java
public class Circulo extends PuntoAbstract{
    public static final double PI = 3.141592;
    private int radio;

    public Circulo(int x, int y, int r) {
        super(x, y);
        radio = r;
    }

    public int getRadio (){
        return radio;
    }

    public void setRadio (int radio) {
        this.radio = radio;
    }

    @Override
    public double calcularArea() {
        return Math.PI * Math.pow(radio, 2);
    }

    @Override
    public double calcularPerimetro(){
        return 2 * Math.PI * radio;
    }

    public String toString(){
        return super.toString() + "\nradio: " + radio;
    }
}
